In [4]:
import sys
from pathlib import Path
import numpy as np
import pandas as pd

project_root = str(Path().absolute().parent)
if project_root not in sys.path:
    sys.path.append(project_root)

from src.validate.TimeGan.dtw import (
    compute_dtw)
from src.validate.TimeGan.fid import (
    compute_fid)


In [13]:
real_data_df = pd.read_csv('/Users/srb/Work/RLCT/data/processed/BTCUSDT_1h_Jan_2025.csv')  
synthetic_data_df = pd.read_csv('/Users/srb/Work/RLCT/data/synthetic/synthetic_data_new.csv')  

# Convert DataFrames to NumPy arrays
real_data = real_data_df.values
synthetic_data = synthetic_data_df.values

dtw_value = compute_dtw(real_data, synthetic_data)
print(f"DTW: {dtw_value}")

fid_value = compute_fid(real_data, synthetic_data)
print(f"FID: {fid_value}")

DTW: 148780747.81016687
FID: 40062877425.91469


## BTCUSDT_1h_Jan_2025.csv && synthetic_data.csv
DTW: 148781089.70037362
FID: 40063293986.20711


## BTCUSDT_1h_Jan_2025.csv && local.csv
DTW: 148780739.60074028
FID: 40062870391.443855

## BTCUSDT_1h_Jan_2025.csv && synthetic_data_new.csv
DTW: 148780747.81016687
FID: 40062877425.91469



